src/main/java/uk/gov/hmcts/cmc/claimstore/documents/ClaimIssueReceiptService.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.cmc.claimstore.documents;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.config.properties.pdf.DocumentTemplates;
+import uk.gov.hmcts.cmc.claimstore.documents.content.ClaimIssueReceiptContentProvider;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.reform.cmc.pdf.service.client.PDFServiceClient;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class ClaimIssueReceiptService {
+
+    private final ClaimIssueReceiptContentProvider contentProvider;
+    private final DocumentTemplates documentTemplates;
+    private final PDFServiceClient pdfServiceClient;
+
+    @Autowired
+    public ClaimIssueReceiptService(
+        ClaimIssueReceiptContentProvider contentProvider,
```

Comment:
I think convention is to make arguments final even for constructors. Please add final modifier.

Reply:
I'd like to challenge that convention then. What purpose does it serve? When used with function arguments, especially if the body is just a couple of lines, does it really help with anything? If it doesn't then it's not justifiable IMO, because then it's just reducing code's signal to noise ratio.